### PR TITLE
Add CMake long double cross-compile defaults

### PR DESCRIPTION
**Pull request by derobins**
HDF5 performs a couple of checks at build time to see if long double values can be converted correctly (IBM's Power architecture uses a special format for long doubles). These checks are performed using TRY_RUN, which is a problem when cross-compiling.

These checks now use default values appropriate for most non-Power systems when cross-compiling. The cache values can be pre-set if necessary, which will preempt both the TRY_RUN and the default.

Affected values:
    H5_LDOUBLE_TO_LONG_SPECIAL      (default no)
    H5_LONG_TO_LDOUBLE_SPECIAL      (default no)
    H5_LDOUBLE_TO_LLONG_ACCURATE    (default yes)
    H5_LLONG_TO_LDOUBLE_CORRECT     (default yes)
    H5_DISABLE_SOME_LDOUBLE_CONV    (default no)

Fixes GitHub #3585

### CODE REVIEW

File: config/cmake/ConfigureChecks.cmake [hunk 1]
```diff
@@ -872,35 +872,46 @@ if (HDF5_BUILD_FORTRAN)
...
     else ()
-      set (${TEST} "" CACHE INTERNAL ${msg})
-      message (VERBOSE "${msg}... no")
-      file (APPEND ${CMAKE_BINARY_DIR}/CMakeFiles/CMakeError.log
-          "Test ${TEST} Compile failed with the following output:\n ${OUTPUT}\n"
-      )
+      # Use the default if there's no cache variable and cross-compiling
+      if ($def)
+        message (VERBOSE "${msg}... yes (cross-compile default)")
+        set (${TEST} 1 CACHE INTERNAL ${msg})
```
Comment (byrnHDF) [inline config/cmake/ConfigureChecks.cmake:909]: Shouldn't you use the value of def? (Which should be ${def})

Comment (byrnHDF) [reply config/cmake/ConfigureChecks.cmake:909]: > Shouldn't you use the value of def? (Which should be ${def})

In line 907

Comment (derobins) [reply config/cmake/ConfigureChecks.cmake:909]: Fixed

**Review by mattjala — APPROVED** [review]

**Comment by nikitalita** [comment]
Yo, while this may not fix windows-x86_64 to arm64 cross-compiling, this DOES fix macos arm64-to-x86_64 and vice-versa.

test repo with vcpkg with a portfile pointed at this PR:  https://github.com/nikitalita/hdf5_test/actions/runs/6860017162/job/18653170927

**Review by bmribler — APPROVED** [review]

**Review by hyoklee — APPROVED** [review]

**Pull request merged**